frontend/src/app/shared/dialog/objective-dialog/objective-form.component.ts

```diff
@@ -74,42 +78,68 @@ export class ObjectiveFormComponent implements OnInit, OnDestroy {
       state: state,
     } as unknown as Objective;
 
-    const submitFunction = this.getSubmitFunction(objectiveDTO.id, objectiveDTO);
-    submitFunction.subscribe((savedObjective: Objective) =>
-      this.closeDialog(savedObjective, false, value.createKeyResults!),
-    );
+    const submitFunction = this.getSubmitFunction(this.data.objective.objectiveId!, objectiveDTO);
+    submitFunction.subscribe((savedObjective: Objective) => {
+      this.closeDialog(savedObjective, false, value.createKeyResults!);
+    });
   }
 
   ngOnInit(): void {
-    const isCreating: boolean = !!this.data.objective.objectiveId;
+    const isEditing: boolean = this.data.objective.objectiveId != undefined;
     this.teams$ = this.teamService.getAllTeams().pipe(takeUntil(this.unsubscribe$));
     this.quarters$ = this.quarterService.getAllQuarters();
     this.currentQuarter$ = this.quarterService.getCurrentQuarter();
-    const objective$ = isCreating
+    this.keyResults$ = isEditing
+      ? this.objectiveService.getAllKeyResultsByObjective(this.data.objective.objectiveId || -1)
+      : of([]);
+    const objective$ = isEditing
       ? this.objectiveService.getFullObjective(this.data.objective.objectiveId!)
       : of(this.getDefaultObjective());
-    forkJoin([objective$, this.quarters$, this.currentQuarter$]).subscribe(([objective, quarters, currentQuarter]) => {
-      this.quarters = quarters;
-      const teamId = isCreating ? objective.teamId : this.data.objective.teamId;
-      const newEditQuarter = isCreating ? currentQuarter.id : objective.quarterId;
-      let quarterId = getValueFromQuery(this.route.snapshot.queryParams['quarter'], newEditQuarter)[0];
-
-      if (currentQuarter && !this.isBacklogQuarter(currentQuarter.label) && this.data.action == 'releaseBacklog') {
-        quarterId = quarters[1].id;
-      }
 
-      this.state = objective.state;
-      this.version = objective.version;
-      this.teams$.subscribe((value) => {
-        this.currentTeam.next(value.filter((team) => team.id == teamId)[0]);
-      });
+    forkJoin([objective$, this.quarters$, this.currentQuarter$, this.keyResults$]).subscribe(
+      ([objective, quarters, currentQuarter, keyResults]: [Objective, Quarter[], Quarter, KeyResultDTO[]]) => {
+        this.handleDataInitialization(objective, quarters, currentQuarter, keyResults, isEditing);
+      },
+    );
+  }
 
-      this.objectiveForm.patchValue({
-        title: objective.title,
-        description: objective.description,
-        team: teamId,
-        quarter: quarterId,
-      });
+  private handleDataInitialization(
+    objective: Objective,
+    quarters: Quarter[],
+    currentQuarter: Quarter,
+    keyResults: KeyResultDTO[],
+    isEditing: boolean,
+  ): void {
+    this.quarters = quarters;
+
+    // Determine the team ID to set in the form: existing team for editing or default team for new objectives
```

Comment:
Nice, now I can finally understand what is happening without reading 20 LOC :+1: